packages/compass-schema/src/modules/create-d3-tip.js

```diff
@@ -0,0 +1,40 @@
+import { css, palette } from '@mongodb-js/compass-components';
+import d3 from 'd3';
+require('./d3-tip')(d3);
+
+const toolTipStyles = (darkMode) => css`
+  z-index: 5;
+  line-height: 1;
+  padding: 8px;
+  background: ${darkMode ? palette.white : palette.black};
+  color: ${darkMode ? palette.black : palette.white};
```

Comment:
Having styles that are not changing in one place and the ones that are changing separate from each other will make refactoring this in the future easier

Reply:
not sure if this is a good or a bad thing I'm doing here 😄 .. ideally I would like to move all the styles where they are used and avoid global classes entirely, but that would require a bit more work

Reply:
I'm just suggesting to have something like (a made up example):

```js
const commonStyles = css`
  position: relative;
  padding: 10px;
`

const darkStyles = css`
  color: black;
`

const lightStyles = css`
  color: white;
`

...

render() {
  return <div className={cx(commonStyles, darkMode ? darkStyles : lightStyles)}>
}
```

instead of having whole style switched because reading through them and figuring out what is the dark mode part there is a bit easier that way

Reply:
Aaahh ok i see now i understand what you mean, that makes sense, how you do that with stuff like `&:after` ? wouldn't those classes be kind-of structurally coupled between each other?

Reply:
Not exactly sure what you mean by that, but you totally can separate theming and layout for pseudo elements same way you can for everything else:

```js
const commonStyles = css`
  position: relative;
  padding: 10px;
  &:after {
    position: absolute;
    content: ""
  }
`

const darkStyles = css`
  color: black;
  &:after {
    color: white;
  }
`

... etc
```

and will have a similar effect of clear separation between theming and layout in the code